File: app/src/main/java/hd/seekbar/MainActivity.java
package hd.seekbar;

import android.os.Bundle;
import android.os.Handler;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import com.hdd.seekbar.DSeekBar;

public class MainActivity extends AppCompatActivity {

    private DSeekBar dSeekBar;
    private Handler handler = new Handler();
    private int i = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dSeekBar = findViewById(R.id.activity_main_dsb_seek_bar);
        dSeekBar.setTotalDuration(180000)
                .setDSeekListener(new DSeekBar.DSeekListener() {
                    @Override
                    public void onChange(int duration, int totalDuration, float percent, String text) {
                        i = duration;
                        Log.e("test", text);
                    }

                    @Override
                    public void onError(String error) {
                        Log.e("test", error);
                    }
                });

        Runnable runnable = new Runnable() {
            @Override
            public void run() {
                dSeekBar.setDuration(i);
                i = i < dSeekBar.getTotalDuration() ? i + 100 : 0;
                handler.postDelayed(this, 100);
            }
        };

        handler.post(runnable);
    }
}

File: app/src/main/java/hd/seekbar/MainActivity.kt
package hd.seekbar

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.hdd.seekbar.DSeekListener
import kotlinx.android.synthetic.main.activity_main.*
import java.util.Timer
import java.util.TimerTask

/**
 * Create on 2/4/21
 * @author duonghd
 */

class MainActivity : AppCompatActivity(R.layout.activity_main) {

    private var isFocus = false
    private var i = 0L

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        dSeekbar.setTotalDuration(60000)
            .setDSeekListener(object : DSeekListener {
                override fun onChange(duration: Long, totalDuration: Long, percent: Float, text: String, isFocus: Boolean) {
                    this@MainActivity.isFocus = isFocus
                    this@MainActivity.i = duration
                }

                override fun onError(error: String) {
                    Log.e("test", error)
                }
            })

        Timer().schedule(
            object : TimerTask() {
                override fun run() {
                    if (!isFocus) {
                        runOnUiThread {
                            dSeekbar.setDuration(i)
                        }
                        i = if (i < dSeekbar.getTotalDuration()) i + 10 else 0
                    }
                }
            },
            0,
            10
        )
    }
}